File: MCP服务器/finance/QuantConnect_mcp-server_3a656b0_python/tests/algorithms/LiveInsights.cs
#region imports
    using System;
    using QuantConnect;
    using QuantConnect.Algorithm.Framework.Alphas;
    using QuantConnect.Algorithm.Framework.Portfolio;
    using QuantConnect.Algorithm;
    using QuantConnect.Data;
#endregion
public class LiveInsightsTestAlgorithm : QCAlgorithm
{
    private int _insights;
    private Symbol _symbol;

    public override void Initialize()
    {
        SetStartDate(2025, 7, 1);
        _insights = 0;
        _symbol = AddCrypto("BTCUSD", Resolution.Second).Symbol;
        SetPortfolioConstruction(new EqualWeightingPortfolioConstructionModel());
    }

    public override void OnData(Slice data)
    {
        // Only emit 10 insights.
        if (_insights++ >= 10)
        {
            return;
        }
        // Determine the direction.
        var direction = Portfolio.Invested ? InsightDirection.Flat : InsightDirection.Up;
        // Emit the insight.
        EmitInsights(Insight.Price(_symbol, TimeSpan.FromDays(1), direction));
    }
}

File: reference-mcp-servers/finance/QuantConnect_mcp-server_3a656b0_python/tests/algorithms/live_insights.py
# region imports
from AlgorithmImports import *
# endregion


class LiveInsightsTestAlgorithm(QCAlgorithm):

    def initialize(self):
        self.set_start_date(2025, 7, 10)
        self._insights = 0
        self._symbol = self.add_crypto('BTCUSD', Resolution.SECOND).symbol
        self.set_portfolio_construction(
            EqualWeightingPortfolioConstructionModel()
        )

    def on_data(self, data: Slice):
        # Only emit 10 insights.
        if self._insights >= 10:
            return
        self._insights += 1
        # Determine the direction.
        if self.portfolio.invested:
            direction = InsightDirection.FLAT
        else:
            direction = InsightDirection.UP
        # Emit the insight.
        self.emit_insights(
            Insight.price(self._symbol, timedelta(1), direction)
        )
